aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1897,34 +1855,18 @@ impl AptosVM {
             arguments,
             &mut gas_meter,
         ) {
-            Ok(result) => {
-                ViewFunctionOutput::new(Ok(result), Self::gas_used(gas_budget, &gas_meter))
+            Ok(result) => ViewFunctionOutput::new(
+                Ok(result),
+                Self::gas_used(max_gas_amount.into(), &gas_meter),
```

Comment:
nit: gas used is computed same way and used in both flows, can we assign to a variable?